src/benchmarklib/benchmark_runner.cpp

```diff
@@ -264,13 +269,19 @@ void BenchmarkRunner::_benchmark_ordered() {
         static_cast<float>(std::chrono::duration_cast<std::chrono::nanoseconds>(_state.benchmark_duration).count());
     const auto duration_seconds = duration_of_all_runs_ns / 1'000'000'000.f;
     const auto items_per_second = static_cast<float>(result.successful_runs.size()) / duration_seconds;
-    const auto num_successful_runs = result.successful_runs.size();
-    const auto duration_per_item =
-        num_successful_runs > 0 ? static_cast<float>(duration_seconds) / static_cast<float>(num_successful_runs) : NAN;
+
+    // Compute mean by using accumulators
+    boost::accumulators::accumulator_set<double, boost::accumulators::stats<boost::accumulators::tag::mean>> acc;
+    for (auto entry : result.successful_runs) {
+      auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(entry.duration);
```

Comment:
What is the benefit of rounding this to ms? Any idea why the numbers reported by the binary and by the comparison script are slightly off?

Reply:
I changed the cast to `nanoseconds`. At the moment I can't find an explanation of why the numbers are slightly off.

Reply:
Do they still differ with the recent changes? I'd expect them to be cause by the previous inaccuracy cause by cutting of  here.

Reply:
If I don't understand that wrong, ms are stored in integers. So we cut of a larger number of nanoseconds. The "live results" are coming from the benchmark runner (which cuts off) and are thus always less-than-equal compared to the python script output (which does a "correct" aggregation).

Reply:
Thank's for the explanation. With the recent change to `nanoseconds ` rounding discrepancies seems to be gone.